src/main/java/apoc/graph/document/builder/DocumentToGraph.java

```diff
@@ -129,11 +175,55 @@ private boolean isSimpleType(Map.Entry<String, Object> e) {
         return true;
     }
 
-    public VirtualGraph create(Collection<Map> coll) {
+    public VirtualGraph create(Collection<Map<String, Object>> coll) {
         Map<String, Set<Node>> nodes = new HashMap<>();
         Set<Relationship> relationships = new LinkedHashSet<>();
         coll.forEach(map -> fromDocument(map, null, null, nodes, relationships));
         return new VirtualGraph("Graph", nodes.values().stream().flatMap(Set::stream).collect(Collectors.toCollection(LinkedHashSet::new)), relationships, Collections.emptyMap());
     }
 
+    public Map<Map<String, Object>, List<Long>> findDuplicates(Collection<Map<String, Object>> documentCollection) {
+        AtomicLong index = new AtomicLong(-1);
+        return documentCollection.stream()
+                .flatMap(e -> {
+                    long lineDup = index.incrementAndGet();
+                    return flatMapFields(e).stream()
+                            .map(ee -> new AbstractMap.SimpleEntry<Map, Long>(ee, lineDup));
+                })
+                .collect(Collectors.groupingBy(Map.Entry::getKey,
+                        Collectors.mapping(Map.Entry::getValue, Collectors.toList())))
+                .entrySet()
+                .stream()
+                .filter(e -> e.getValue().size() > 1)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private List<Map<String, Object>> flatMapFields(Map<String, Object> map) {
```

Comment:
can just return a stream if you use it always as streams :)
and concat the `Stream.of(map).cconcat(map.values().stream().filter(...)).flatMap())` with to combine the streams

Reply:
done